### PR TITLE
gpinitstandby respects checksums

### PR DESCRIPTION
**Pull request by larham**
Add a behave test to show that gpinitstandby uses the same data_checksum as master.

-- add pipeline task for gpinitstandby
-- introduce template to DRY some pipeline tasks for behave

### CODE REVIEW

File: gpMgmt/test/behave/mgmt_utils/environment.py [hunk 1]
```diff
@@ -82,3 +82,13 @@ def after_scenario(context, scenario):
     if 'gpssh' in context.feature.tags:
         run_command(context, 'sudo tc qdisc del dev lo root netem')
 
+    # for cleaning up after @given('"{path}" has its permissions set to "{perm}"')
```
Comment (Chibin) [inline gpMgmt/test/behave/mgmt_utils/environment.py]: 👍

**Comment by Chibin** [comment]
Consider adding this job to the Release Candidate?
 https://github.com/larham/gpdb/blob/c4d8ba9141a4ea12b4be7e9ea928bf65afea29ef/concourse/pipelines/pipeline.yml#L1541

@pivotal-mike How do we determine which jobs goes to the Release Candidate tab?

File: concourse/pipelines/pipeline.yml [hunk 2]
```diff
@@ -849,36 +829,25 @@ jobs:
...
-      passed: [compile_gpdb_centos6]
-      trigger: true
-    - get: centos-gpdb-dev-6
+  - aggregate: *gets_for_behave
   - task: gpinitsystem
     file: gpdb_src/concourse/tasks/behave_gpdb.yml
     image: centos-gpdb-dev-6
     params:
-      BEHAVE_FLAGS: --tags=gpinitsystem
+      BEHAVE_FLAGS: gpinitsystem
```
Comment (Chibin) [inline concourse/pipelines/pipeline.yml]: Should this be `BEHAVE_FLAGS: --tags=gpinitsystem`?

File: gpMgmt/test/behave/mgmt_utils/gpinitstandby.feature [hunk 3]
```diff
@@ -2,12 +2,44 @@
...
         And the segments are synchronized
-        And "$GPHOME/share" has "555" permissions
+        And "$GPHOME/share" has its permissions set to "555"
+        When the user initializes a standby on the same host as master
+        Then gpinitstandby should return a return code of 0
+        And gpinitstandby should not print "Traceback" to stdout
+        And rely on environment.py to restore path permissions
+
+    Scenario: gpinitstandby creates the standby with default data_checksums on (standby on same host as master)
+        Given the database is running
```
Comment (Chibin) [inline gpMgmt/test/behave/mgmt_utils/gpinitstandby.feature]: There's an assumption here that we're starting the database with default options.

File: gpMgmt/test/behave/mgmt_utils/steps/mgmt_utils.py [hunk 4]
```diff
@@ -2641,6 +2642,50 @@ def delete_data_dir(host):
...
+    hostname = get_master_hostname('template1')[0][0]
+    port_guaranteed_open = get_open_port()
+    temp_data_dir = tempfile.mkdtemp() + "/standby_datadir"
+    cmd = "gpinitstandby -a -s %s -P %d -F pg_system:%s" % (hostname, port_guaranteed_open, temp_data_dir)
+    run_gpcommand(context, cmd)
+    context.standby_data_dir = temp_data_dir
+    context.standby_was_initialized = True
+
+
+# from https://stackoverflow.com/questions/2838244/get-open-tcp-port-in-python/2838309#2838309
```
Comment (Chibin) [inline gpMgmt/test/behave/mgmt_utils/steps/mgmt_utils.py]: 👍

File: gpMgmt/test/behave/mgmt_utils/steps/mgmt_utils.py [hunk 5]
```diff
@@ -2641,6 +2642,50 @@ def delete_data_dir(host):
...
+    if not os.path.exists(path):
+        raise Exception('Path does not exist! "%s"' % path)
+    old_permissions = os.stat(path).st_mode  # keep it as a number that has a meaningful representation in octal
+    test_permissions = int(perm, 8)          # accept string input with octal semantics and convert to a raw number
+    os.chmod(path, test_permissions)
+    context.path_for_which_to_restore_the_permissions = path
+    context.permissions_to_restore_path_to = old_permissions
+
+
+@then('rely on environment.py to restore path permissions')
```
Comment (Chibin) [inline gpMgmt/test/behave/mgmt_utils/steps/mgmt_utils.py]: 👍

File: gpMgmt/test/behave/mgmt_utils/gpinitstandby.feature (hunk 3 above)
Comment (larham) [inline gpMgmt/test/behave/mgmt_utils/gpinitstandby.feature]: yes, and this assumption is verified  within that scenario, on line 22:

`        Then gpconfig should print "Master  value: on" to stdout`

File: concourse/pipelines/pipeline.yml (hunk 2 above)
Comment (larham) [inline concourse/pipelines/pipeline.yml]: Thanks, fixed

**Pull request merged**